lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -194,7 +194,7 @@ sub search_by_email {
 
     my $query = 'SELECT ' . $self->_columns .
         ' FROM ' . $self->_table .
-        q" WHERE regexp_replace(regexp_replace(email, '(\+.*@)', '@'), '\.(.*@)', '\1', 'g') ~ ?" .
+        q"WHERE (regexp_replace(regexp_replace(email, '[@+].*', ''), '\.', '', 'g') || regexp_replace(email, '.*@', '@')) ~ ?" .
```

Comment:
I'm not sure what each bit here is doing (not sure if it is complex or I am slow). Mind explaining step by step? :)

Reply:
This is explained in commit message (with a few typos):
It splits `email` value on `@` character to process separately:
* user info (to remove `+suffix` and periods `.`),
* domain name (no change).

Is it more clear like that?

Reply:
OH. Duh. I was thinking of || as OR, not concatenation. Sorry.

Reply:
Step-by-step example with `email = 'a.b.c+d@f.g'`:
1. `regexp_replace('a.b.c+d@f.g', '[@+].*', '')` ⇒ `'a.b.c'`
2. `regexp_replace('a.b.c', '\.', '', 'g')` ⇒ `'a'`
3. `regexp_replace('a.b.c+d@f.g', '.*@', '@')` ⇒ `'@f.g'`
4. `'a' || '@f.g'` ⇒ `'a@f.g'`

Reply:
Thanks! Seems sensible.

Reply:
Updated commit message to fix typo (s/prefix/suffix/) and mention contatenation operator `||`: https://github.com/metabrainz/musicbrainz-server/pull/2102/commits/06d4763535f34db8287f1a82dd7aff9fb03caf11.